tianshou/data/batch.py

```diff
@@ -17,7 +18,14 @@
 import numpy as np
 import torch
 
-IndexType = slice | int | np.ndarray | list[int]
+_SingleIndexType = slice | int | EllipsisType
+IndexType = (
+    np.ndarray
+    | _SingleIndexType
+    | list[_SingleIndexType]
+    | tuple[_SingleIndexType, ...]
+    | list[int]
```

Comment:
this shouldn't be necessary, already covered by `list[_SingleIndexType]`

Reply:
I can remove it now. There are no issues when not explicitly defining `list[int]`. 

Previously, mypy was complaining because in `Collector._reset_state`, it couldn't infer that `list[_SingleIndexType]` includes `list[int]` too.

Reply:
Sounds good